publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java

```diff
@@ -533,13 +535,27 @@ public void shutdownCompleted(ShutdownSignalException cause) {
 
     /**
      * This method is used to give random channel
+     * 
      * @return channel
      * @throws RemRemPublishException
      */
     private Channel giveMeRandomChannel() throws RemRemPublishException {
-        if ((rabbitConnection == null || !rabbitConnection.isOpen())) {
-            createRabbitMqConnection();
+        if (rabbitConnection == null || !rabbitConnection.isOpen()) {
+            // If the rabbitConnection became null and if shutdown is not initiated by application
+            // the call will go to if block.
+            if (shutdown) {
+                try {
+                    Thread.sleep(5000);
+                    shutdown = false;
+                } catch (InterruptedException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
```

Comment:
What will happen if the sleep is interrupted after 4.5 sec? Should 'shutdown' keep its previous value?

Reply:
I guess it will contains the previous value, Or may be we can add **shutdown = true** as the first line in try block 
try {
shutdown = true;
Thread.sleep(5000);
} catch { ... }